File: src/main/scala/com/vc/medium/L531.scala
package com.vc.medium

object L531 {
  def findLonelyPixel(picture: Array[Array[Char]]): Int = {
    val n = picture.length
    val m = picture(0).length

    val row = new Array[Int](n)
    val col = new Array[Int](m)

    picture.indices.foreach(i => {
      picture(i).indices.foreach(j => {
        if(picture(i)(j) == 'B') {
          row(i) += 1
          col(j) += 1
        }
      })
    })

    var res = 0
    picture.indices.foreach(i => {
      picture(i).indices.foreach(j => {
        if(picture(i)(j) == 'B') {
          if(row(i) == 1 && col(j) == 1) res += 1
        }
      })
    })
    res
  }
}

File: src/main/java/com/vc/medium/L531.java
package com.vc.medium;

class L531 {
    public int findLonelyPixel(char[][] picture) {
        if(picture == null || picture.length == 0) return 0;

        int n = picture.length;
        int m = picture[0].length;

        int[] row = new int[n];
        int[] col = new int[m];

        for(int i = 0; i < n; i++) {
            for(int j = 0; j < m; j++) {
                if(picture[i][j] == 'B') {
                    row[i]++;
                    col[j]++;
                }
            }
        }

        int res = 0;
        for(int i = 0; i < n; i++) {
            for(int j = 0; j < m; j++) {
                if(picture[i][j] == 'B' && row[i] == 1 && col[j] == 1) {
                    res++;
                }
            }
        }

        return res;
    }
}
